mono/metadata/sre.c

```diff
@@ -2803,7 +2803,8 @@ reflection_methodbuilder_to_mono_method (MonoClass *klass,
 	m->slot = -1;
 	m->flags = rmb->attrs;
 	m->iflags = rmb->iattrs;
-	m->name = mono_string_to_utf8_image_ignore (image, rmb->name);
+	MonoStringHandle name = MONO_HANDLE_NEW (MonoString, rmb->name);
```

Comment:
This may leak a handle because `reflection_methodbuilder_to_mono_method` isn't wrapped in `HANDLE_FUNCTION_ENTER/RETURN`.

Also nearly every single thing in the `ReflectionMethodBuilder` struct is a managed object.  We really need to approach `sre.c` more systematically - changing one thing at a time in here is just going to get confusing.

In this particular instance I'd use a static helper function to temporarily wrap `rmb->name` in a handle and do the conversion.  Also leave a `FIXME` comment to come back here and get rid of the helper when we're ready to update `sre.c`

Reply:
I'm not sure if sre.c needs to be done all at once or not. I think the individual changes can be reasoned about, though I agree I didn't get it right -- no enter/exit macros. I'll fix in some fashion now.